### PR TITLE
systemd: use text.template to generate mount unit

### PR DESCRIPTION
**Pull request by mardy**
Thanks @mvo for the suggestion! :-)

### CODE REVIEW

**Review by zyga — APPROVED** [review]
Looks good, nice and clean

File: systemd/systemd.go [hunk 1]
```diff
@@ -1152,24 +1161,17 @@ func writeMountUnitFile(u *MountUnitOptions) (mountUnitName string, err error) {
...
-	mountUnitName, err = filepath.Base(mu), osutil.AtomicWriteFile(mu, []byte(content), 0644, 0)
+	outf, err := osutil.NewAtomicFile(mu, 0644, 0, osutil.NoChown, osutil.NoChown)
 	if err != nil {
-		return "", err
+		return "", fmt.Errorf("cannot open mount unit file: %v", err)
 	}
-	return mountUnitName, nil
+	defer outf.Cancel()
+
+	if err := parsedMountUnitTemplate.Execute(outf, &u); err != nil {
```
Comment (mvo5) [inline systemd/systemd.go:1171]: This is very nice!

**Review by mvo5 — APPROVED** [review]
Nice, thank you!

**Review by MiguelPires — APPROVED** [review]
LGTM

**Pull request merged**